base/compiler/ssair/passes.jl

```diff
@@ -952,6 +952,7 @@ function pattern_match_typeof(compact::IncrementalCompact, typ::DataType, fidx::
 end
 
 function refine_new_effects!(𝕃ₒ::AbstractLattice, compact::IncrementalCompact, idx::Int, stmt::Expr)
+    iszero(compact[SSAValue(idx)][:flag] & IR_FLAG_NOTHROW) || return # already accurate
```

Comment:
Is there no case where we can refine either of the other flags here?

Reply:
Maybe? It looks like `new_expr_effect_flags` always returns `consistent=false` (, which I guess seems to be a bug itself), so there is no case when it is refined, but `IR_FLAG_EFFECT_FREE` can be refined here especially because we have the additional `pattern_match_typeof` helper.